### PR TITLE
remove protoc dep

### PR DESCRIPTION
**Pull request by S1nus**
This is a hideous hack to remove `protoc` build dependency. Do not merge, but we need to solve this

### CODE REVIEW

File: .DS_Store
Comment (fl0rek) [inline .DS_Store:1]: please remove

**Comment by oblique** [comment]
I don't really like this, but if we gonna do it I really prefer move the `build.rs` code in a crate called `proto-gen` that will have [publishing disabled](https://doc.rust-lang.org/cargo/reference/manifest.html#the-publish-field). And that crate must set the [`prost_build::Config::out_dir`](https://docs.rs/prost-build/latest/prost_build/struct.Config.html#method.out_dir) to `proto/src/out`.

**Comment by dimazhornyk** [comment]
+1 to the importance of this, dependency on `protoc` means that anyone who imports your crates has to have protoc installed, which is quite unnecessary in most cases
e.g. our CI when importing your crate: https://github.com/matter-labs/zksync-era/actions/runs/11112479333/job/30874671258?pr=2983#step:8:739

the option that @oblique suggested is the easiest one, but you might also consider [protox](https://github.com/andrewhickman/protox), it's a great alternative that lets you avoid the `protoc` dependency

**Comment by S1nus** [comment]
Protox looks like an easy compromise, https://github.com/eigerco/lumina/pull/402, let me know thoughts on 402

**Pull request closed without merging**